afl/src/lib.rs

```diff
@@ -18,6 +19,206 @@ unsafe extern "C" {
     static __afl_fuzz_ptr: *const u8;
 }
 
+// AFL++ IJON functions in afl-compiler-rt
+unsafe extern "C" {
+    pub fn ijon_max(addr: u32, val: u64);
+    pub fn ijon_min(addr: u32, val: u64);
+    pub fn ijon_set(addr: u32, val: u32);
+    pub fn ijon_inc(addr: u32, val: u32);
+    pub fn ijon_xor_state(val: u32);
+    pub fn ijon_reset_state();
+    pub fn ijon_simple_hash(x: u64) -> u64;
+    pub fn ijon_hashint(old: u32, val: u32) -> u32;
+    pub fn ijon_hashstr(old: u32, val: *const c_char) -> u32;
+    pub fn ijon_hashmen(old: u32, val: *const u8, len: usize) -> u32;
+    pub fn ijon_hashstack_backtrace() -> u32;
+    pub fn ijon_hashstack() -> u32;
+    pub fn ijon_strdist(a: *const u8, b: *const u8) -> u32;
+    pub fn ijon_memdist(a: *const u8, b: *const u8, len: usize) -> u32;
+    pub fn ijon_max_variadic(addr: u32, ...);
+    pub fn ijon_min_variadic(addr: u32, ...);
+}
+
+#[macro_export]
+macro_rules! ijon_inc {
+    ($x:expr) => {{
+        use std::sync::atomic::{AtomicU32, Ordering};
+        static LOC_CACHE: AtomicU32 = AtomicU32::new(0);
+        let mut loc = LOC_CACHE.load(Ordering::Relaxed);
+        if loc == 0 {
+            let cfile = CString::new(file!()).unwrap();
+            let new_val = unsafe { ijon_hashstr(line!(), cfile.as_ptr()) };
```

Comment:
```suggestion
            let new_val = unsafe { afl::ijon_hashstr(line!(), cfile.as_ptr()) };
```
Like with the `CString` change, please make a similar change everywhere an `ijon` identifier is used in a macro.

This will be a PITA because I know there are lots of them.